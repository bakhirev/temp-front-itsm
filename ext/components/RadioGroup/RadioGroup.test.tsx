import React from 'react';
import { shallow } from 'enzyme';

import { RadioGroup } from './';

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
}));

describe('RadioGroup', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const radioGroupProps = {
    onChange: jest.fn(),
    value: 'Moscow',
    list: [
      { label: 'Москва', value: 'Moscow' },
      { label: 'Самара', value: 'Samara' },
      { label: 'Воронеж', value: 'Voronez' },
    ],
  };

  it('should render component with required properties', () => {
    const wrapper = shallow(<RadioGroup {...radioGroupProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label', () => {
    const wrapper = shallow(<RadioGroup {...radioGroupProps} label="RadioGroup" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small size', () => {
    const wrapper = shallow(<RadioGroup {...radioGroupProps} size={'small'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = shallow(<RadioGroup {...radioGroupProps} disabled />);
    expect(wrapper).toMatchSnapshot();
  });
});
