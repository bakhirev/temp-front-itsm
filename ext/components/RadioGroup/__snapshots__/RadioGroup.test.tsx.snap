// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render component with disabled state 1`] = `
<styled.div
  size="big"
>
  <Component
    checked={true}
    disabled={true}
    error={false}
    key="0"
    label="Москва"
    onChange={[Function]}
    size="big"
    value="Moscow"
  />
  <Component
    checked={false}
    disabled={true}
    key="1"
    label="Самара"
    onChange={[Function]}
    size="big"
    value="Samara"
  />
  <Component
    checked={false}
    disabled={true}
    key="2"
    label="Воронеж"
    onChange={[Function]}
    size="big"
    value="Voronez"
  />
</styled.div>
`;

exports[`RadioGroup should render component with label 1`] = `
<styled.div
  size="big"
>
  <Styled(styled.div)
    size="big"
    theme={Object {}}
  >
    RadioGroup
  </Styled(styled.div)>
  <Component
    checked={true}
    error={false}
    key="0"
    label="Москва"
    onChange={[Function]}
    size="big"
    value="Moscow"
  />
  <Component
    checked={false}
    key="1"
    label="Самара"
    onChange={[Function]}
    size="big"
    value="Samara"
  />
  <Component
    checked={false}
    key="2"
    label="Воронеж"
    onChange={[Function]}
    size="big"
    value="Voronez"
  />
</styled.div>
`;

exports[`RadioGroup should render component with required properties 1`] = `
<styled.div
  size="big"
>
  <Component
    checked={true}
    error={false}
    key="0"
    label="Москва"
    onChange={[Function]}
    size="big"
    value="Moscow"
  />
  <Component
    checked={false}
    key="1"
    label="Самара"
    onChange={[Function]}
    size="big"
    value="Samara"
  />
  <Component
    checked={false}
    key="2"
    label="Воронеж"
    onChange={[Function]}
    size="big"
    value="Voronez"
  />
</styled.div>
`;

exports[`RadioGroup should render component with small size 1`] = `
<styled.div
  size="small"
>
  <Component
    checked={true}
    error={false}
    key="0"
    label="Москва"
    onChange={[Function]}
    size="small"
    value="Moscow"
  />
  <Component
    checked={false}
    key="1"
    label="Самара"
    onChange={[Function]}
    size="small"
    value="Samara"
  />
  <Component
    checked={false}
    key="2"
    label="Воронеж"
    onChange={[Function]}
    size="small"
    value="Voronez"
  />
</styled.div>
`;
